benchmarks/src/main/java/io/crate/operation/aggregation/IntervalAggregationBenchmark.java

```diff
@@ -0,0 +1,156 @@
+    private OffHeapMemoryManager offHeapMemoryManager;
+    private AggregateCollector onHeapCollector;
+    private AggregateCollector offHeapCollector;
+
+    @Setup
+    public void setUp() throws Exception {
+        final InputCollectExpression inExpr0 = new InputCollectExpression(0);
+        Functions functions = new ModulesBuilder()
+            .add(new AggregationImplModule())
+            .createInjector().getInstance(Functions.class);
+
+        final IntervalSumAggregation intervalSumAggregation = (IntervalSumAggregation) functions.getQualified(
+                Signature.aggregate(
+                    IntervalSumAggregation.NAME,
+                    DataTypes.INTERVAL.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature()),
+                List.of(DataTypes.INTERVAL),
+                DataTypes.INTERVAL
+            );
+
+        onHeapMemoryManager = new OnHeapMemoryManager(bytes -> {});
+        offHeapMemoryManager = new OffHeapMemoryManager();
+        onHeapCollector = new AggregateCollector(
+            Collections.singletonList(inExpr0),
+            RamAccounting.NO_ACCOUNTING,
+            onHeapMemoryManager,
+            Version.CURRENT,
+            AggregateMode.ITER_FINAL,
+            new AggregationFunction[] { intervalSumAggregation },
+            Version.CURRENT,
+            new Input[][] { { inExpr0 } },
+            new Input[] { Literal.BOOLEAN_TRUE }
+        );
+        offHeapCollector = new AggregateCollector(
+            Collections.singletonList(inExpr0),
+            RamAccounting.NO_ACCOUNTING,
+            offHeapMemoryManager,
+            Version.CURRENT,
+            AggregateMode.ITER_FINAL,
+            new AggregationFunction[] { intervalSumAggregation },
+            Version.CURRENT,
+            new Input[][] { { inExpr0 } },
+            new Input[] { Literal.BOOLEAN_TRUE }
+        );
+    }
+
+    @TearDown
+    public void tearDown() {
+        onHeapMemoryManager.close();
+        offHeapMemoryManager.close();
+    }
+
+    @Benchmark
+    public Iterable<Row> benchmarkIntervalAggregationOnHeap() {
+        Object[] state = onHeapCollector.supplier().get();
+        BiConsumer<Object[], Row> accumulator = onHeapCollector.accumulator();
+        Function<Object[], Iterable<Row>> finisher = onHeapCollector.finisher();
+        for (int i = 0; i < rows.size(); i++) {
+            accumulator.accept(state, rows.get(i));
+        }
+        return finisher.apply(state);
+    }
+
+    @Benchmark
+    public Iterable<Row> benchmarkIntervalAggregationOffHeap() {
+        Object[] state = offHeapCollector.supplier().get();
+        BiConsumer<Object[], Row> accumulator = offHeapCollector.accumulator();
+        Function<Object[], Iterable<Row>> finisher = offHeapCollector.finisher();
+        for (int i = 0; i < rows.size(); i++) {
+            accumulator.accept(state, rows.get(i));
+        }
+        return finisher.apply(state);
+    }
```

Comment:
```suggestion
    @Benchmark
    public Iterable<Row> benchmarkIntervalAggregation() {
        Object[] state = onHeapCollector.supplier().get();
        BiConsumer<Object[], Row> accumulator = onHeapCollector.accumulator();
        Function<Object[], Iterable<Row>> finisher = onHeapCollector.finisher();
        for (int i = 0; i < rows.size(); i++) {
            accumulator.accept(state, rows.get(i));
        }
        return finisher.apply(state);
    }
```

There is no off-heap support for the sum-interval aggregation.

Fine to keep the other one, but for comparison/regression detection, adding a case in https://github.com/crate/crate-benchmarks/blob/uservisits-insert-bulk/specs/aggregate.toml might be the better option, as we've more tooling in place for that and it's unlikely that we'll keep two versions around for the jmh benchmark to have a baseline/comparison

Reply:
I thought so, but would it be better to add a spec there use 2 stored timestamps and subtraction to produce an interval, since interval is not a stored type, or should we use `unnest` with literal intervals?

Reply:
thx!